### PR TITLE
Fix/membership request notification

### PR DESCRIPTION
**Pull request by gevorgmansuryan**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
[Cannot delete notification for "join space" request](https://github.com/humhub/humhub/issues/6241)

### CODE REVIEW

**Comment by ArchBlood** [comment]
Not sure the use of the word `actual` is the right one needed here, using `active` seems more useful here.

File: protected/humhub/modules/notification/components/BaseNotification.php [hunk 1]
```diff
@@ -129,6 +129,16 @@ protected function category()
         return null;
     }
 
+    /**
+     * Checks if notification is still actual before sending
+     *
+     * @return bool
+     */
+    public function isActual()
```
Comment (luke-) [inline protected/humhub/modules/notification/components/BaseNotification.php:137]: I also think we should improve the wording here.

What do you think about `$notification->isValid()`. This method should check if the notification should still exists, if not, the notification should be deleted.

File: protected/humhub/modules/space/notifications/ApprovalRequest.php [hunk 2]
```diff
@@ -78,6 +79,18 @@ public function category()
         return new SpaceMemberNotificationCategory;
     }
 
+    /**
+     * @inerhitdoc +
```
Comment (luke-) [inline protected/humhub/modules/space/notifications/ApprovalRequest.php:83]: `@inheritdoc`

File: protected/humhub/modules/notification/components/BaseNotification.php (hunk 1 above)
Comment (luke-) [reply protected/humhub/modules/notification/components/BaseNotification.php:137]: Please also add the as PHPDoc `@since 1.15`

**Review by luke- — CHANGES_REQUESTED** [review]
Thanks, I think the approach fixes the bug. Can you scan for other Notifications or Situations which may have a similar problem?

**Review by luke- — APPROVED** [review]

**Pull request merged**